### ISSUE TITLE
Revise API for geom_forecast()

### ISSUE THREAD

**Issue by earowang** [tidyverts/fabletools#38]
Ideally,

```r
geom_forecast(
  mapping = aes(x = x, y = y, distribution = ..distribution..), data = fbl_ts,
  levels = c(80, 95), stats = "identity", other_args
)
```

* `x` & `y` inherit from `ggplot()`
* `levels` better to be constants instead of aesthetic mapping.
* `stats = "identity"` better to be the default instead of  "forecast"

**Comment by mitchelloharawild** [comment]
I don't know how this can be done whilst continuing to support the legends for confidence level (and point forecast type, https://github.com/tidyverts/fabletools/issues/163).

Further, I don't think fabletools is specific to distributions now, with this functionality reimplemented in [distributional](https://github.com/mitchelloharawild/distributional).

Rather I think it is useful for distributional to gain support for plotting distributions using geoms, as suggested in https://github.com/mitchelloharawild/distributional/issues/6

Depending on context, a distribution can be handled discretely (comparing two distributions) or continuously (forecasts), so I don't know how this translates to ggplot.

For now, I think it is best to further the functionality of autoplot<fbl_ts> and autolayer<fbl_ts> and separate the ribbon-style interval plotting and the point forecasts into two separate layers. I think <hilo> objects can reasonably be considered as continuous, and so a [`geom_hilo_ribbon()`](https://github.com/mitchelloharawild/distributional/issues/11) can be written to power plotting of intervals, whilst `geom_line()` should suffice for point forecasts.

Learnings/challenges from [gghdr](https://github.com/ropenscilabs/gghdr) would be useful for this.

**Comment by mitchelloharawild** [comment]
I'm now thinking that `geom_forecast()` is not so useful (but still should be handled similarly to `geom_smooth()`). Instead, I think this functionality should be `geom_distribution()` and exist in the distributional package as it is not specific to time series or forecasting. `geom_distribution()` seems pretty complicated to integrate with ggplot2, and so I've started by adding `geom_hilo_ribbon()` for plotting intervals. The `autolayer<fbl_ts>()` method now uses `geom_hilo_ribbon()` for intervals and `geom_line()` for point forecasts, which feels better in terms of the input data structures for each geom. Further development in plotting distributions should be done in the distributional package, for example: https://github.com/mitchelloharawild/distributional/issues/6

Closing as `geom_forecast()` is no longer needed to produce the functionality suggested for this API revision.
